src/ResourceManagement/PrivateDns/Domain/VirtualNetworkLink/Definition/IDefinition.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.PrivateDns.Fluent.VirtualNetworkLink.Definition
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The entirety of the virtual network link definition.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IDefinition<ParentT> :
+        IBlank<ParentT>,
+        IWithAttach<ParentT>
+    {
+    }
+
+    /// <summary>
+    /// The first stage of a virtual network link definition.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IBlank<ParentT> :
+        IWithAttach<ParentT>
+    {
+    }
+
+    /// <summary>
+    /// The final stage of the virtual network link definition.
+    /// At this stage, any remaining optional settings can be specified, or the virtual network link
+    /// definition can be attached.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithAttach<ParentT> :
+        ResourceManager.Fluent.Core.ChildResource.Definition.IInDefinition<ParentT>,
+        IWithAutoRegistration<ParentT>,
+        IWithETagCheck<ParentT>,
+        IWithReferenceVirtualNetwork<ParentT>,
+        IWithRegion<ParentT>,
+        IWithTags<ParentT>
+    {
+    }
+
+    /// <summary>
+    /// The stage of the virtual network link definition allowing to manage auto-registration of virtual network records.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithAutoRegistration<ParentT>
+    {
+        /// <summary>
+        /// Specifies to enable auto-registration for virtual network records.
+        /// </summary>
+        /// <return>The next stage of the definition.</return>
+        IWithAttach<ParentT> EnableAutoRegistration();
+
+        /// <summary>
+        /// Specifies to disable auto-registration for virtual network records.
+        /// </summary>
+        /// <return>The next stage of the definition.</return>
+        IWithAttach<ParentT> DisableAutoRegistration();
+    }
+
+    /// <summary>
+    /// The stage of the virtual network link definition allowing to enable ETag validation.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithETagCheck<ParentT>
+    {
+        /// <summary>
+        /// Specifies that If-None-Match header needs to set to  to prevent updating an existing virtual network link.
```

Comment:
@yaohaizh  Any good idea?

Reply:
Right, we need make sure the etag meaning here. @xccc-msft, please confirm this from the spec. 

@weidongxu-microsoft , could you help design the etag handling in a big scope?

Reply:
@yaohaizh Currently the existing interface for user is consistent. WithETagCheck() enables implicit check (* for create, resource.ETag for update). WithETagCheck(string eTag) for explicit check (user specifies the exact eTag for update).
In future we would like to see if we can do less in coding, let generator or certain base class to do the work.

Here my concern is on the docstring. Current one seems not clear. Actually the sentence likely broken with missing '\*', since the actual implemenation is "If-None-Match header needs to set to '\*'".

Maybe just
> Specifies If-None-Match header to prevent ...

woudl be better?

Reply:
Done for the change